### ISSUE TITLE
WMT Pytorch does not hit target 

### ISSUE THREAD

**Issue by priyakasimbeg** [mlcommons/algorithmic-efficiency#275]
WMT pytorch did not hit target 2/2 times when tested.

## Description
WMT pytorch workload was ran 2 times and did not hit the target.
The runs were either ran till the target was hit or until the max time  was reached. 
Target bleu score = 30.6446
Max time = 80,000s
Target total number of steps = 100,000

The info for each run actual bleu score and total number of steps:
- run 1: 181719 steps, val bleu score 29.400
- run 2: 181,656 steps, val bleu score 29.7179

## Steps to Reproduce
Commit: 6ee99c (Dec 15)
sacrebleu==1.3.1

On kasimbeg-6 run:
```
torchrun --redirects 1:0,2:0,3:0,4:0,5:0,6:0,7:0 --standalone --nnodes=1 --nproc_per_node=8 submission_runner.py \
    --framework=pytorch \
    --data_dir=/home/kasimbeg/tensorflow_datasets \
    --experiment_dir=/home/kasimbeg \
    --experiment_name=target_setting \
    --workload=wmt \
    --submission_path=reference_algorithms/target_setting_algorithms/pytorch_adamw.py \
    --tuning_search_space=reference_algorithms/target_setting_algorithms/wmt/tuning_search_space.json --profile
```

## Source or Possible Fix
We previously had issues with the sacrebleu version but I believe those were reconciled and sacrebleu was set to 1.3.1. So there might be something else at play here. We should also confirm the final target numbers.

**Comment by runame** [comment]
I have also noticed that the PyTorch and the Jax WMT workloads miss the target by a small margin (see the thread in PR #270). But it should be much closer than what you have seen; I suspect you report the bleu score after all steps and not the max validation bleu score reached during training? If you check between 85k and 100k steps, there should be a point where the target is almost reached. Afterwards it goes a bit down again.

Maybe the target is set too ambitiously, or there is still a discrepancy between this repo and the code used for the target setting runs.

**Comment by runame** [comment]
I have adjusted the validation bleu score target from 30.6446 to 30.6 and ran the target setting run twice. In the first run it hit the target after 88375 steps (here it actually hit about 30.668) and in the second after 96938 steps (here it got about 30.624).

**Comment by priyakasimbeg** [comment]
Ah ok yes, I did report the final score and not the max score.  You're right the max score was much better (about 30.52 around 80K steps for this run)
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/12614254/211426163-17bca95e-f2d3-495d-9304-b615db875b78.png">

>> I have adjusted the validation bleu score target from 30.6446 to 30.6 and ran the target setting run twice. In the first run it hit the target after 88375 steps (here it actually hit about 30.668) and in the second after 96938 steps (here it got about 30.624).
Ok that is good, are both of those for pytorch?

**Comment by priyakasimbeg** [comment]
I compared the profiles of the jax runs and the validation scores seem to flatline at about 100K, which is probably expected since the step hint is around 100K steps. 
The pytorch numbers deteriorate, potentially indicating a diff in the learning rate scheduler between pytorch and jax or some other diff.

**Comment by runame** [comment]
> Ok that is good, are both of those for pytorch?
yes.

**Comment by priyakasimbeg** [comment]
For completeness, since there was some question on overfitting. I don't think this is overfitting because the training metrics also decrease:
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/12614254/211580237-0b68cc05-37a1-4393-a5f5-fc1dcda6c9df.png">

**Comment by priyakasimbeg** [comment]
Ran 4 more runs of this and the distribution looks ok now:

![image](https://user-images.githubusercontent.com/12614254/215870815-71c1273e-b146-4ed6-95cf-c94a928c2458.png)
